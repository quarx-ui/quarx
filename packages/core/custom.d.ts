declare module '*.png';
declare module '*.md';
declare module '*.svg';
declare module '*.svg' {
    // eslint-disable-next-line import/newline-after-import
    import { SFC, SVGProps } from 'react';
    export const ReactComponent: SFC<SVGProps<SVGSVGElement>>;
    const src: string;
    export default src;
}

interface ObjectConstructor {
    keys<T>(obj: T): Array<Exclude<keyof T, symbol | number>>;
    entries<T>(obj: T): Array<[Exclude<keyof T, symbol | number>, T[keyof T]]>;
}
