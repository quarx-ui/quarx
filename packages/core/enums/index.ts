export * from './QxDevice';
export * from './QxSize';
export * from './QxBorderSize';
