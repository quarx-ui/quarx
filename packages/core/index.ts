import './polyfills';

// WARNING: Изменение порядка экспорта может сломать сборку
export * from './types';
export * from './utils';
export * from './enums';
export * from './styles';
export * from './src';
