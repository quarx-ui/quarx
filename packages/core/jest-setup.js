import '@testing-library/jest-dom';
import 'mock-match-media/jest-setup.cjs';
import { setMedia } from 'mock-match-media';
import { cleanup } from '@testing-library/react';

setMedia({
    'prefers-color-scheme': 'light',
});
global.beforeEach(() => {
    cleanup();
    jest.clearAllTimers();
    jest.clearAllMocks();
});
