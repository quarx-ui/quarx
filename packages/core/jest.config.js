module.exports = {
    setupFilesAfterEnv: ['<rootDir>/jest-setup.js'],
    transform: {
        '^.+\\.(js|jsx|ts|tsx)$': [
            'babel-jest',
            {
                rootMode: 'upward',
            },
        ],
    },
    // eslint-disable-next-line no-useless-escape
    modulePathIgnorePatterns: ['.*\.test\.pw\..*'],
    testEnvironment: 'jest-environment-jsdom',
};
