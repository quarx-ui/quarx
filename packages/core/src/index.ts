export * from './styled/Badge';
export * from './styled/Button';
export * from './styled/Checkbox';
export * from './styled/Counter';
export * from './styled/Link';
export * from './styled/Loader';
export * from './styled/RadioButton';
export * from './styled/Selection';
export * from './styled/Switcher';
export * from './styled/Tabs';
export * from './styled/TextField';
