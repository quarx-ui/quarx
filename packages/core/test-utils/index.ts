export { expectRootCn } from './expectRootCn';
export * from './testBEM';
