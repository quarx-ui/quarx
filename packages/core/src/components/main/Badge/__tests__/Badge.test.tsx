import { render, screen } from '@testing-library/react';
import { Badge, BadgeProps, BadgeStyleParams } from '@core';
import { testStyleParams } from '@quarx-ui/core/test-utils';

describe('Badge', () => {
    testStyleParams<BadgeStyleParams, BadgeProps>(
        Badge,
        {
            size: 'medium',
            borderRadius: 'max',
            color: 'brand',
            type: 'contained',
        },
        { children: 'Статус' },
    )({
        color: ['brand', 'secondary', 'info', 'success', 'warning', 'danger', 'text'],
        size: ['small', 'medium', 'large'],
        borderRadius: ['xSmall', 'small', 'medium', 'large', 'xLarge', 'max'],
        type: ['contained', 'outlined', 'ghosted'],
    });

    it('with counter', () => {
        const { asFragment } = render(<Badge counter={55}>Статус</Badge>);

        expect(screen.queryByText('Статус')).toBeInTheDocument();
        expect(screen.queryByText('55')).toBeInTheDocument();
        expect(asFragment()).toMatchSnapshot();
    });
});
