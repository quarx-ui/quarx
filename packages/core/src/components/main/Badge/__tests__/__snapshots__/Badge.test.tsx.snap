// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Badge style params borderRadius large 1`] = `
<DocumentFragment>
  <div
    class="QxBadge QxBadge_type_contained QxBadge_size_medium QxBadge_borderRadius_large QxBadge_color_brand css-vag3g8-QxBadge-root"
    data-testid="testStyleParams-borderRadius-large"
  >
    Статус
  </div>
</DocumentFragment>
`;

exports[`Badge style params borderRadius max 1`] = `
<DocumentFragment>
  <div
    class="QxBadge QxBadge_type_contained QxBadge_size_medium QxBadge_borderRadius_max QxBadge_color_brand css-be4r60-QxBadge-root"
    data-testid="testStyleParams-borderRadius-max"
  >
    Статус
  </div>
</DocumentFragment>
`;

exports[`Badge style params borderRadius medium 1`] = `
<DocumentFragment>
  <div
    class="QxBadge QxBadge_type_contained QxBadge_size_medium QxBadge_borderRadius_medium QxBadge_color_brand css-uz2fw4-QxBadge-root"
    data-testid="testStyleParams-borderRadius-medium"
  >
    Статус
  </div>
</DocumentFragment>
`;

exports[`Badge style params borderRadius small 1`] = `
<DocumentFragment>
  <div
    class="QxBadge QxBadge_type_contained QxBadge_size_medium QxBadge_borderRadius_small QxBadge_color_brand css-mulw5v-QxBadge-root"
    data-testid="testStyleParams-borderRadius-small"
  >
    Статус
  </div>
</DocumentFragment>
`;

exports[`Badge style params borderRadius xLarge 1`] = `
<DocumentFragment>
  <div
    class="QxBadge QxBadge_type_contained QxBadge_size_medium QxBadge_borderRadius_xLarge QxBadge_color_brand css-s14g5f-QxBadge-root"
    data-testid="testStyleParams-borderRadius-xLarge"
  >
    Статус
  </div>
</DocumentFragment>
`;

exports[`Badge style params borderRadius xSmall 1`] = `
<DocumentFragment>
  <div
    class="QxBadge QxBadge_type_contained QxBadge_size_medium QxBadge_borderRadius_xSmall QxBadge_color_brand css-174t3cb-QxBadge-root"
    data-testid="testStyleParams-borderRadius-xSmall"
  >
    Статус
  </div>
</DocumentFragment>
`;

exports[`Badge style params color brand 1`] = `
<DocumentFragment>
  <div
    class="QxBadge QxBadge_type_contained QxBadge_size_medium QxBadge_borderRadius_max QxBadge_color_brand css-be4r60-QxBadge-root"
    data-testid="testStyleParams-color-brand"
  >
    Статус
  </div>
</DocumentFragment>
`;

exports[`Badge style params color danger 1`] = `
<DocumentFragment>
  <div
    class="QxBadge QxBadge_type_contained QxBadge_size_medium QxBadge_borderRadius_max QxBadge_color_danger css-1r656fk-QxBadge-root"
    data-testid="testStyleParams-color-danger"
  >
    Статус
  </div>
</DocumentFragment>
`;

exports[`Badge style params color info 1`] = `
<DocumentFragment>
  <div
    class="QxBadge QxBadge_type_contained QxBadge_size_medium QxBadge_borderRadius_max QxBadge_color_info css-92oby5-QxBadge-root"
    data-testid="testStyleParams-color-info"
  >
    Статус
  </div>
</DocumentFragment>
`;

exports[`Badge style params color secondary 1`] = `
<DocumentFragment>
  <div
    class="QxBadge QxBadge_type_contained QxBadge_size_medium QxBadge_borderRadius_max QxBadge_color_secondary css-noa5o4-QxBadge-root"
    data-testid="testStyleParams-color-secondary"
  >
    Статус
  </div>
</DocumentFragment>
`;

exports[`Badge style params color success 1`] = `
<DocumentFragment>
  <div
    class="QxBadge QxBadge_type_contained QxBadge_size_medium QxBadge_borderRadius_max QxBadge_color_success css-hsyuow-QxBadge-root"
    data-testid="testStyleParams-color-success"
  >
    Статус
  </div>
</DocumentFragment>
`;

exports[`Badge style params color text 1`] = `
<DocumentFragment>
  <div
    class="QxBadge QxBadge_type_contained QxBadge_size_medium QxBadge_borderRadius_max QxBadge_color_text css-jo7dk9-QxBadge-root"
    data-testid="testStyleParams-color-text"
  >
    Статус
  </div>
</DocumentFragment>
`;

exports[`Badge style params color warning 1`] = `
<DocumentFragment>
  <div
    class="QxBadge QxBadge_type_contained QxBadge_size_medium QxBadge_borderRadius_max QxBadge_color_warning css-7fal5f-QxBadge-root"
    data-testid="testStyleParams-color-warning"
  >
    Статус
  </div>
</DocumentFragment>
`;

exports[`Badge style params size large 1`] = `
<DocumentFragment>
  <div
    class="QxBadge QxBadge_type_contained QxBadge_size_large QxBadge_borderRadius_max QxBadge_color_brand css-wipob6-QxBadge-root"
    data-testid="testStyleParams-size-large"
  >
    Статус
  </div>
</DocumentFragment>
`;

exports[`Badge style params size medium 1`] = `
<DocumentFragment>
  <div
    class="QxBadge QxBadge_type_contained QxBadge_size_medium QxBadge_borderRadius_max QxBadge_color_brand css-be4r60-QxBadge-root"
    data-testid="testStyleParams-size-medium"
  >
    Статус
  </div>
</DocumentFragment>
`;

exports[`Badge style params size small 1`] = `
<DocumentFragment>
  <div
    class="QxBadge QxBadge_type_contained QxBadge_size_small QxBadge_borderRadius_max QxBadge_color_brand css-1lamplc-QxBadge-root"
    data-testid="testStyleParams-size-small"
  >
    Статус
  </div>
</DocumentFragment>
`;

exports[`Badge style params type contained 1`] = `
<DocumentFragment>
  <div
    class="QxBadge QxBadge_type_contained QxBadge_size_medium QxBadge_borderRadius_max QxBadge_color_brand css-be4r60-QxBadge-root"
    data-testid="testStyleParams-type-contained"
  >
    Статус
  </div>
</DocumentFragment>
`;

exports[`Badge style params type ghosted 1`] = `
<DocumentFragment>
  <div
    class="QxBadge QxBadge_type_ghosted QxBadge_size_medium QxBadge_borderRadius_max QxBadge_color_brand css-1l53dnr-QxBadge-root"
    data-testid="testStyleParams-type-ghosted"
  >
    Статус
  </div>
</DocumentFragment>
`;

exports[`Badge style params type outlined 1`] = `
<DocumentFragment>
  <div
    class="QxBadge QxBadge_type_outlined QxBadge_size_medium QxBadge_borderRadius_max QxBadge_color_brand css-1h88krq-QxBadge-root"
    data-testid="testStyleParams-type-outlined"
  >
    Статус
  </div>
</DocumentFragment>
`;

exports[`Badge with counter 1`] = `
<DocumentFragment>
  <div
    class="QxBadge QxBadge_type_contained QxBadge_size_medium QxBadge_borderRadius_max QxBadge_color_brand css-be4r60-QxBadge-root"
  >
    Статус
    <span
      class="QxCounter QxCounter_type_white QxCounter_size_medium QxCounter_color_brand QxBadge-rightItem QxBadge-item css-1d607ij-QxCounter-root-QxBadge-rightItem-QxBadge-item-Badge"
    >
      55
    </span>
  </div>
</DocumentFragment>
`;
